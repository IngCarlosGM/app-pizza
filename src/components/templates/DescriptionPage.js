import React from 'react'
import PropTypes from 'prop-types'

export const DescriptionPage = ({ description, myStyle= "" }) => {
    return (
        <div className={`description-page ${ myStyle }`}>
            { description }
        </div>
    )
}

DescriptionPage.propTypes = {
    description: PropTypes.string.isRequired,
    myStyle: PropTypes.string
}
