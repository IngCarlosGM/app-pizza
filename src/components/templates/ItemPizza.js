import React from 'react';
import PropTypes from 'prop-types';

export const ItemPizza = ({ itemPizza }) => {

    return (
        <div className="container-item-store pointer animate__bounceIn">
            <div className="item-store">
                <img 
                    className="img-item-store"
                    src={ `../assets/pizzas/pizza-${itemPizza.id}.png` } 
                    alt="Logo pizza" 
                />
            </div>
            <div className="flex-column">
                <span className="title-item-store">{ itemPizza.name }</span>
            </div>
        </div>
    )
}

ItemPizza.propTypes = {
    itemPizza: PropTypes.object.isRequired
}