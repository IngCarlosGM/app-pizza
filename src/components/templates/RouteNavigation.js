import React from 'react';
import PropTypes from 'prop-types';

export const RouteNavigation = ({ route }) => {
    return (
        <span className="route-navigation">
            { route }
        </span>
    )
}

RouteNavigation.propTypes = {
    route: PropTypes.string.isRequired
}