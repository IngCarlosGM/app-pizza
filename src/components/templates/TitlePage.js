import React from 'react'
import PropTypes from 'prop-types'

export const TitlePage = ({ title, myStyle= "" }) => {
    return (
        <h1 className={` ${ myStyle }`}>
            { title }
        </h1>
    )
}

TitlePage.propTypes = {
    title: PropTypes.string.isRequired,
    myStyle: PropTypes.string
}