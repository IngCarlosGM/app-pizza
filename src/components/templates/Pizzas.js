import React from 'react';

import { ItemPizza } from './ItemPizza';

export const Pizzas = ({ pizzas }) => {

    return (
        <div className="grid-pizzas mt-third">
            {
                pizzas && pizzas.map( itemPizza => {
                    return <ItemPizza 
                                key={ `itemPizza${ itemPizza.id }` } 
                                itemPizza={ itemPizza }
                            />
                })
            }
        </div>
    )
}