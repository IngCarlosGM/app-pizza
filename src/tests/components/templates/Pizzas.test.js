import { shallow } from "enzyme";
import { Pizzas } from "../../../components/templates/Pizzas";
import dataApp from '../../data/dataApp.json';

describe('Testing ItemPizza component', () => {

    const pizzas = dataApp.response.stores[0].products;

    const wrapper = shallow(
        <Pizzas 
            pizzas= { pizzas }
        />
    );
   
    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should to have pizzas', () => {
       expect( wrapper.find('ItemPizza').length ).toBe( pizzas.length );
    });
    
    
});
