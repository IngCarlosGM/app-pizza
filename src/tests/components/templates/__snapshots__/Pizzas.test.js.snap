// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ItemPizza component should render correctly 1`] = `
<div
  className="grid-pizzas mt-third"
>
  <ItemPizza
    itemPizza={
      Object {
        "id": 1,
        "name": "Pizza de Pollo",
      }
    }
    key="itemPizza1"
  />
  <ItemPizza
    itemPizza={
      Object {
        "id": 2,
        "name": "Piza Pollo y Champiñones",
      }
    }
    key="itemPizza2"
  />
  <ItemPizza
    itemPizza={
      Object {
        "id": 3,
        "name": "Pizza Vegetales",
      }
    }
    key="itemPizza3"
  />
  <ItemPizza
    itemPizza={
      Object {
        "id": 4,
        "name": "Pizza Vegetales y Atun ",
      }
    }
    key="itemPizza4"
  />
  <ItemPizza
    itemPizza={
      Object {
        "id": 5,
        "name": "Pizza 3 Quesos con Champiñones",
      }
    }
    key="itemPizza5"
  />
  <ItemPizza
    itemPizza={
      Object {
        "id": 6,
        "name": "Pizza Jamon y Queso",
      }
    }
    key="itemPizza6"
  />
</div>
`;
